xformers/ops/fmha/triton.py

```diff
@@ -3,63 +3,430 @@
 # This source code is licensed under the BSD license found in the
 # LICENSE file in the root directory of this source tree.
 
+"""
```

Comment:
IIUC, this file is to implement the flash attention here rather than importing from `third_party/flash-attention/flash_attn/flash_attn_triton.py`, is that right? Is there any amd specific optimization here (btw it's great to import more masks than just the lowerTriangular) or it's similar?

Reply:
Yes, this is from a patch PRed to `http://github.com/ROCmSoftwarePlatform/xformers` by Grigorysizeov@meta.com. 

But we did not keep the PR records because we just re-build the repo.  

I will re-submit PR from `http://github.com/ROCmSoftwarePlatform/xformers`  branch rather than from my personal repo.

Thanks